src/kernels/kernels.jl

```diff
@@ -30,7 +30,7 @@ each column is an observation.
 """
 function cov(k::Kernel, X₁::AbstractMatrix, X₂::AbstractMatrix, kerneldata::KernelData=EmptyData())
     n1, n2 = size(X₁, 2), size(X₂, 2)
-    cK = Array{eltype(X₁)}(undef, n1, n2)
+    cK = Array{eltype(X₂)}(undef, n1, n2)
```

Comment:
I'm guessing this is more useful for your usecase, but it doesn't quite generalise. A better solution might be to compute a pilot value `cov_ij(k, X₁, X₂, 42, 43, dim)` and use its type to build cK. Or maybe just `promote(X₁[1,1], X₂[1,1])` or `promote_type(typeof(X₁), typeof(X₂))`?

Reply:
Yes.